server/game/cards/attachments/03/motley.js

```diff
@@ -0,0 +1,41 @@
+const DrawCard = require('../../../drawcard.js');
+
+class Motley extends DrawCard {
+    setupCardAbilities(ability) {
+        this.whileAttached({
+            effect: ability.effects.addTrait('Fool')
+        });
+        this.reaction({
+            when: {
+                onAttackersDeclared: (event, challenge) => challenge.isAttacking(this.parent)
+            },
+            handler: () => {
+                this.parent.controller.discardAtRandom(1);
+                this.game.addMessage('{0} uses {1} to discard 1 card at random from {2}\'s hand', 
+                                      this.controller, this, this.parent.controller);
+            }
+        });
+        this.reaction({
+            when: {
+                onDefendersDeclared: (event, challenge) => challenge.isDefending(this.parent)
```

Comment:
These two don't need to be defined as two separate reactions. You can define it as one reaction and have the `when` contain both event. See Catelyn Stark WotN for an example.